### DISCUSSION TITLE
Bitwise | in Permission Classes Isn't Working Properly

### DISCUSSION THREAD

**Discussion by decaf-dev** [encode/django-rest-framework#8131] [Potential Issue]
I have found a weird behavior in my custom permissions that I have setup for my user model view set. I have a model view set with the methods: GET, PUT, and PATCH. The goal of my behavior is to allow an administrator to GET, PUT, and PATCH for any User object. I also want a User to be able to GET, PUT and PATCH ONLY if their id matches the id of the User object that they're trying to edit, basically if they're the owner.

![image](https://user-images.githubusercontent.com/40307803/129473194-bced288b-4ec3-4270-82d7-170b1e23ca4d.png)
![image](https://user-images.githubusercontent.com/40307803/129473210-07c27537-c3e8-4adc-b036-a2e0256dae98.png)
![image](https://user-images.githubusercontent.com/40307803/129473231-f94908f5-9500-4e4f-897a-610f4c032ef0.png)

Here are my permissions:

![image](https://user-images.githubusercontent.com/40307803/129473252-764321b7-bdda-4c3b-abab-bf1024bab950.png)

The issue is that when I use the & operator to combine permissions, Django seems to ignore the 2nd operation and default to True, even though the permission resulted in false

![image](https://user-images.githubusercontent.com/40307803/129473268-c3d98d27-77db-4966-b287-14100840816f.png)
![image](https://user-images.githubusercontent.com/40307803/129473287-5651c6e3-853b-44a0-bbac-5308196b8f74.png)

As you can see, the isAdmin class returned False and the isUser class was never checked, but the patch method wasn't denied.

Now, if I remove the isUser permission, it will still evaluate False but it will restrict permissions and deny the patch method.

![image](https://user-images.githubusercontent.com/40307803/129473325-091d63f5-74ca-4757-9395-9cb7c0081edc.png)
![image](https://user-images.githubusercontent.com/40307803/129473333-829c0fa8-e7c5-4ada-bb8b-3a9947496146.png)
![image](https://user-images.githubusercontent.com/40307803/129473349-720fb53c-dabb-4d68-9654-03d8ebdcb158.png)

I believe this is a bug because Django is not evaluating the 2nd permission, the isUser permission, when it should. I have tried reversing the order of the permissions and the same thing happens. My authentication is fine and the IsAuthenticated permission works great.

Please let me know if you need more information

**Comment by SpookyGlitches** [comment]
Hi, any updates on this?  I'm trying to do the same where admins and object owners are able to do safe and unsafe methods while other users are only allowed to do safe methods.

I have 2 permission on my class too and the 2nd class permission will not be called (I call ```print``` too to check if it's called). 

To visualize, here's the row in our good old friend truth table where it is wrong:

| CanModifyOrDelete (custom permission) | IsAdminUser | CanModifyOrDelete \| IsAdminUser | Expected Behavior | Actual Behavior |
|---------------------------------------|-------------|----------------------------------|-------------------|-----------------|
| F                                     | F           | F                                | Unauthorized      | Authorized      |

Reversing the order to ```IsAdminUser | CanModifyOrDelete``` also has the same results as above.

My custom permission ```CanModifyOrDelete``` and the ```IsAdminUser``` provided by the framework works fine too. So yeah, maybe it's the bitwise ```|```  that isn't working properly?

_Update: Found an issue disccusion here https://github.com/encode/django-rest-framework/issues/7117_
